pkg/healthcheck/grpc.go

```diff
@@ -0,0 +1,68 @@
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/runconduit/conduit/controller/gen/common"
+	"google.golang.org/grpc"
+)
+
+type grpcStatusChecker interface {
+	SelfCheck(ctx context.Context, in *pb.SelfCheckRequest, opts ...grpc.CallOption) (*pb.SelfCheckResponse, error)
```

Comment:
this has to match `SelfCheck()` as defined in `api.proto`. is there a tighter way to couple this? i guess a similar question with `CheckResult` defined in `common.proto` and `healthcheck.go`?

Reply:
I can't think of anything. My idea was that having this being unit tested and using the structural type should be good enough, but maybe I'm missing something?